source/extensions/tracers/opentelemetry/tracer.cc

```diff
@@ -122,8 +122,41 @@ void Span::setAttribute(absl::string_view name, const OTelAttribute& attribute_v
   *span_.add_attributes() = key_value;
 }
 
+::opentelemetry::proto::trace::v1::Status_StatusCode
+convertGrpcStatusToTraceStatusCode(::opentelemetry::proto::trace::v1::Span_SpanKind kind,
+                                   absl::string_view value) {
+  uint64_t grpc_status_code;
+  if (!absl::SimpleAtoi(value, &grpc_status_code)) {
+    return ::opentelemetry::proto::trace::v1::Status::STATUS_CODE_UNSET;
+  }
```

Comment:
if this logic okay? invalid code treat as `CODE_UNSET`?

Reply:
TBH, I don't know, so keep as before behavior(UNSET).

Reply:
Then, please add a comment to explain this case let the future developer/reader know it's not by design.